packages/components/src/components/checkbox/checkbox.jsx

```diff
@@ -3,73 +3,80 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import Icon from '../icon';
 
-const Checkbox = ({
-    className,
-    classNameLabel,
-    disabled,
-    id,
-    label,
-    defaultChecked,
-    onChange, // This needs to be here so it's not included in `otherProps`
-    value,
-    withTabIndex,
-    ...otherProps
-}) => {
-    const [checked, setChecked] = React.useState(defaultChecked || value);
-    const input_ref = React.useRef();
+const Checkbox = React.forwardRef(
+    (
+        {
+            className,
+            classNameLabel,
+            disabled,
+            id,
+            label,
+            defaultChecked,
+            onChange, // This needs to be here so it's not included in `otherProps`
+            value,
+            withTabIndex,
+            ...otherProps
+        },
+        ref
+    ) => {
+        const [checked, setChecked] = React.useState(defaultChecked || value);
+        const input_ref = React.useRef(ref || null);
```

Comment:
if `ref` has a value, then we might have to use `input_ref.current.current` 🤔

Reply:
I changed this to `React.useRef(ref?.current || null);`

Reply:
I don't think it works that way. We need to set the forwarded ref here.

if we use setRef function we can set both `input_ref` and forwarded `ref` to the input element

```
setRef = (el_input) => {
      ref.current = input_ref.current = el_input;
}
```
and
```
<input ref={setRef} />
```

Reply:
@balakrishna-binary  Thank you for your comment. I changed it